sopel/modules/url.py

```diff
@@ -222,12 +226,14 @@ def check_callbacks(bot, trigger, url, run=True):
 
 
 def find_title(url, verify=True):
+    parser = TitleParser()
     try:
-        r = requests.get(url, verify=verify, headers=default_headers)
-        r.encoding = 'utf-8'
-        parser = TitleParser()
-        parser.feed(r.text)
-    except:
+        with requests.get(url, verify=verify, headers=default_headers, timeout=2, stream=True) as r:
+            r.encoding = 'utf-8'
+            chunk = next(r.iter_content(max_bytes, decode_unicode=True))
+            parser.feed(chunk)
```

Comment:
nice! i didn't realise `feed` actually did what it amplied. :)

could we also abort parsing as soon as we found the title instead of necessarily hitting max_bytes?

Reply:
It's certainly possible, I'll take a look in a few hours at making it abort when the parser has found it, pulling smaller chunks.